src/containers/User/UserRegister.tsx

```diff
@@ -1,166 +1,18 @@
-import React, { useState } from 'react'
-import { Button, Checkbox, Form, Input, Message, Segment } from 'semantic-ui-react'
-import {
-  CreateUserPayload,
-  useCreateUserMutation,
-  UsersConnection,
-} from '../../utils/generated/graphql'
-import addNewUser from '../../utils/graphql/fragments/addNewUser.fragment'
-
-interface Snackbar {
-  showMessage: boolean
-  messageTitle: string
-  messageText: string
-  isErrorMessage: boolean
-}
-
-interface User {
-  username: string
-  password: string
-  email: string
-}
+import React from 'react'
+import { useHistory } from 'react-router'
```

Comment:
Could you use the `useRouter` instead? There all these hooks are combined in one place. I think it's easier to have it all contained in one place like that. You can destruct the `push` to be used here.

Reply:
Got it. I had trouble using it before, but I think I get how it works now.

Reply:
Hey @nmadruga , The match, history, location and params properties weren't defined in the `RouterResult` interface in `useRouter`, so using them caused a type error. I've added them in, but can't figure out what their type should be other than `any`. Any ideas?